packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -804,4 +716,160 @@ export class JobService {
+    await jobEntity.save();
+
+    return true;
+  }
+
+  public async processEscrowCancellation(jobEntity: JobEntity): Promise<string> {
+      const { chainId, escrowAddress } = jobEntity;
+
+      const signer = this.web3Service.getSigner(chainId);
+      const escrowClient = await EscrowClient.build(signer);
+      
+      const escrowStatus = await escrowClient.getStatus(escrowAddress);
+      if (escrowStatus === EscrowStatus.Complete || escrowStatus === EscrowStatus.Paid || escrowStatus === EscrowStatus.Cancelled) {
+          this.logger.log(ErrorEscrow.InvalidStatusCancellation, JobService.name);
+          throw new BadRequestException(ErrorEscrow.InvalidStatusCancellation);
+      }
+
+      const balance = await escrowClient.getBalance(escrowAddress);
+      if (balance.eq(0)) {
+          this.logger.log(ErrorEscrow.InvalidBalanceCancellation, JobService.name);
+          throw new BadRequestException(ErrorEscrow.InvalidBalanceCancellation);
+      }
+
+      const transactionHash = await escrowClient.cancel(escrowAddress);
+      return transactionHash;
+  }
+
+  public async refundCronJob() {
+    // TODO: Add retry policy and process failure requests https://github.com/humanprotocol/human-protocol/issues/334
+    const jobEntity = await this.jobRepository.findOne(
+      {
+        status: JobStatus.TO_REFUND,
+        retriesCount: LessThanOrEqual(JOB_RETRIES_COUNT_THRESHOLD),
+        waitUntil: LessThanOrEqual(new Date())
+      },
+      {
+        order: {
+            waitUntil: SortDirection.ASC,
+        },
+      }
+    );
+    if (!jobEntity) return;
+
+    const { id, userId, chainId, escrowAddress, fundAmount, latestTransactionHash } = jobEntity
+
+    const signer = this.web3Service.getSigner(chainId);
+    const transactionReceipt = await signer.provider.getTransactionReceipt(latestTransactionHash);
+
+    if (!transactionReceipt) {
+      this.logger.log(ErrorJob.TransactionNotFound, JobService.name);
+      throw new BadRequestException(ErrorJob.TransactionNotFound);
+    }
+
+    if (transactionReceipt.confirmations < TX_CONFIRMATION_TRESHOLD) {
+      this.logger.log(ErrorJob.TransactionHasNotEnoughAmountOfConfirmations, JobService.name);
+      throw new BadRequestException(ErrorJob.TransactionHasNotEnoughAmountOfConfirmations);
+    }
+
+    const escrowClient = await EscrowClient.build(signer);
+    const tokenAddress = await escrowClient.getTokenAddress(escrowAddress);
+    
+    let refundAmount = new Decimal(fundAmount);
+    if (escrowAddress) {
+      refundAmount = await this.calculateRefundAmount(chainId, tokenAddress, escrowAddress);
+    }
+    
+    await this.paymentService.createRefundPayment({ refundAmount: refundAmount.toNumber(), userId: userId, jobId: id });
+  
+    jobEntity.status = JobStatus.CANCELED;
+    await jobEntity.save();
+
+    return true;
+  }
+
+  public async calculateRefundAmount(chainId: ChainId, tokenAddress: string, escrowAddress: string) {
```

Comment:
@portuu3

Reply:
Here we are working with logs got from EOA. I will try to check, can we get something from cancel method tx.